.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Build
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: 'zulu'
+
+      - uses: actions/cache@v2
```

Comment:
This is supported by setup-java as well: https://github.com/actions/setup-java#caching-packages-dependencies

Reply:
Yes, it is. Don't count on that feature to be permanent, as you may have noticed in this issue https://github.com/actions/setup-java/issues/40#issuecomment-999730509

Reply:
I am reporter of https://github.com/actions/setup-java/issues/40 and this is only about configuring Maven but not about caching at all (particularly not about caching Gradle artifacts)

Reply:
I'm well aware of that which is why I typed "as you may have noticed". IMHO `setup-java` should be concerned with just that: setting up a suitable JDK/JRE/JR. Adding caching for Maven or any other Java build tool is beyond its scope.

Reply:
This may happen, but never in a 2.x release of setup-java (is it would be backwards-incompatible). As long as you use setup-java@v2 I would recommend getting rid of additional caching action steps and leverage what is shipped with setup-java@v2 as having additional steps
# make syour yml larger
# doesn't provide additional functionality
# defers builds as every step is additional overhead